src/framework/control.ml

```diff
@@ -625,13 +625,16 @@ struct
            let node_values = LHT.enum lh |> map (Tuple2.map1 fst) in (* drop context from key *)
            let hashtbl_size = if fast_count node_values then count node_values else 123 in
            let by_loc, by_node = Hashtbl.create hashtbl_size, NodeH.create hashtbl_size in
-           node_values |> iter (fun (node, v) ->
-               let loc = Node.location node in
-               (* join values once for the same location and once for the same node *)
-               let join = Option.some % function None -> v | Some v' -> Spec.D.join v v' in
-               Hashtbl.modify_opt loc join by_loc;
-               NodeH.modify_opt node join by_node;
-             );
+           iter (fun (node, v) ->
+              let loc = match node with
+                | Statement s -> Cil.get_stmtLoc s.skind (* nosemgrep: cilfacade *) (* Must use CIL's because syntactic search is in CIL. *)
+                | FunctionEntry _ | Function _ -> Node.location node
+              in
+              (* join values once for the same location and once for the same node *)
+              let join = Option.some % function None -> v | Some v' -> Spec.D.join v v' in
+              Hashtbl.modify_opt loc join by_loc;
+              NodeH.modify_opt node join by_node;
+            ) node_values;
```

Comment:
Does this still behave correctly for the transformations? Could you elaborate on why this change was needed?

Reply:
02bba9350b61590653b87a89e79b68c7fd905ba1 was needed to fix the GobView CI for syntactic/semantic search.

Really, it always needed to be that: since the syntactic search and its location matching are implemented in CIL, then it uses `Cil.get_stmtLoc` as its locations. Goblint's `Node.location` prefers expression locations instead, so the two don't necessarily match.
I don't really know why the communication between CIL and Goblint for syntactic search uses locations, not `sid`-s or something.